tensorflow/cc/gradients/math_grad.cc

```diff
@@ -38,7 +38,15 @@ REGISTER_NO_GRADIENT_OP("NotEqual");
 REGISTER_NO_GRADIENT_OP("LogicalAnd");
 REGISTER_NO_GRADIENT_OP("LogicalOr");
 REGISTER_NO_GRADIENT_OP("LogicalNot");
+// These are really zero (almost everywhere) gradients, but we return
+// NoGradient() following the current python implementation.
+// Additional discussion on this topic:
+// https://github.com/tensorflow/tensorflow/issues/897
+// https://github.com/tensorflow/tensorflow/issues/783
 REGISTER_NO_GRADIENT_OP("Floor");
+REGISTER_NO_GRADIENT_OP("Ceil");
```

Comment:
Can we have a separate test for these as well?

Reply:
Sure thing - I've added a few for the ones I added, as well as for "Floor" which was missing.

(Apologies for also rolling in the formatting diffs - looks like some code got in without clang-format, and it was lazier to fix it together with this change.)